pybamm/simulation.py

```diff
@@ -438,10 +437,13 @@ def solve(self, t_eval=None, solver=None, check_model=True, **kwargs):
             # Re-initialize solution, e.g. for solving multiple times with different
             # inputs without having to build the simulation again
             self._solution = None
+            previous_num_subsolutions = 0
             # Step through all experimental conditions
             inputs = kwargs.get("inputs", {})
             pybamm.logger.info("Start running experiment")
             timer = pybamm.Timer()
+
+            phases = []
```

Comment:
might be useful to add a comment about what a "phase" is. is the number of phases the same as the numbers of steps?

Reply:
I agree. I am not too sure what it means, so a comment would be helpful.

Reply:
I'm thinking one cycle would consist of e.g. "discharge phase", "CC phase", "CV phase". Would you prefer "step" instead?